site/content/entry/citra-progress-report-2023-q2/index.md

```diff
@@ -0,0 +1,548 @@
+
+{{< sidebyside "image" ""
+    "digimonbroke.png=Where are you little monster?"
+    "digimonfix.png=Oh, here!" >}}
+
+Next, let's talk about Disney. Not the company, but a pretty obscure game they made for the Nintendo 3DS called Disney Art Academy, where you get to draw Mickey Mouse and his friends using the touchscreen and stylus. Given the nature of the mechanics of the game, where the use of the touchscreen and stylus is centralized, not many people choose to emulate this title. So we were surprised when a bug report came in, stating that the game had become completely broken after the first refactor.
+
+{{< figure src="disneybroke.png"
+    title="This will haunt our dreams..." >}}
+
+After some digging, GPUCode found out that the new mipmap implementation was the culprit that robbed Mickey of his face. To go into more detail, texture downloads were not adjusted to account for multi-level surfaces and thus readbacks of these surfaces did not function correctly. Rendering and reading back from multi-level surfaces is quite unusual and not something we often see in games. Regardless, we are happy it was reported to us so that we could promptly fix it!
+
+{{< figure src="disneyfix.png"
+    title="Mickey got his face back!" >}}
+
+Alright, what about the performance improvement then? Indeed, by changing the data structure used to store surfaces, Citra is now able to cache and search for them more quickly and efficiently. The old interval map was ditched in favour of a simple page table. Memory locality has also been improved by taking [inspiration](https://github.com/citra-emu/citra/blob/master/src/common/slot_vector.h) from [yuzu](https://yuzu-emu.org/)'s texture cache. All in all, this change should yield a couple of extra frames in games that perform many cache operations and especially in cases where interval map lookups would [slow](https://github.com/citra-emu/citra/pull/5183) things down.
+
+### Add MMPX texture filter ([#6564](https://github.com/citra-emu/citra/pull/6564)) by [stuken](https://github.com/stuken)
+
+What's better than 5 texture filters? Well, how about 6! Introducing the MMPX texture filter, implemented by first time contributor, [stuken](https://github.com/stuken).
+MMPX is a texture filter centered around the preservation of those classic pixel art styles we typically see in 8- and 16-bit era video games. Sprites, fonts, you name it, it’s been given a sweet pixel art style with this new texture filter!
+
+This new texture filter can be enabled in `Emulation -> Configure -> Graphics -> Renderer` (`Citra -> Preferences` on macOS).
+
+{{< single-title-imgs-compare
+    ""
+    "./none.png"
+    "./mmpx.png"
+    >}} 
```

Comment:
Is there a reason a caption was withheld? Are captions broken for the juxtapose shortcode on Citra?

Reply:
There were some problems, but we fixed them yesterday. It just needs to be pushed to this branch.
Also the image size needs to be decreased.

Reply:
Just had a few issues when moving over to the new system, but all fixed now and captions can be added 👍